### PR TITLE
rocm-cmake: changed test API from old to new

### PR DESCRIPTION
**Pull request by AcriusWinter**
Update standalone test API. See below comment for test output.

https://spack.readthedocs.io/en/latest/packaging_guide.html#stand-alone-tests

### CODE REVIEW

**Comment by AcriusWinter** [comment]
==> [2024-06-27-15:25:16.133682] 'make' '-j16'
==> [2024-06-27-15:25:16.231668] 'make' '-j16' 'clean'
PASSED: RocmCmake::test_cmake
==> [2024-06-27-15:25:16.330543] Completed testing
==> [2024-06-27-15:25:16.330843] 
====================== SUMMARY: rocm-cmake-6.1.1-45uzliy =======================
RocmCmake::test_cmake .. PASSED
============================= 1 passed of 1 parts ==============================
============================== 1 passed of 1 spec ==============================

File: var/spack/repos/builtin/packages/rocm-cmake/package.py [hunk 1]
```diff
@@ -63,15 +63,15 @@ def cache_test_sources(self):
...
-            return
+            raise SkipTest("Package must be installed as version @5.5.1 or later")
         test_dir = join_path(self.test_suite.current_test_cache_dir, self.test_src_dir)
         with working_dir(test_dir, create=True):
-            cmake_bin = join_path(self.spec["cmake"].prefix.bin, "cmake")
             prefixes = ";".join([self.spec["rocm-cmake"].prefix])
             cc_options = ["-DCMAKE_PREFIX_PATH=" + prefixes, "."]
-            self.run_test(cmake_bin, cc_options)
+            exe = which(self.spec["cmake"].prefix.bin.cmake)
+            exe(*cc_options)
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/rocm-cmake/package.py]:
```suggestion
            cmake = which(self.spec["cmake"].prefix.bin.cmake)
            cmake(*cc_options)
```

**Review by tldahlgren — DISMISSED** [review]

**Comment by tldahlgren** [comment]
Looks like there are conflicts that need to be resolved.

**Comment by tldahlgren** [comment]
Outputs I'm getting are:

```
$ spack -v test run rocm-cmake
==> Spack test 2e2njki24ibttua7fksuasinvsjr4scv
==> Testing package rocm-cmake-5.4.3-245fk42
...
==> [2024-07-09-16:19:43.363472] test: test_cmake: Test cmake
...
==> [2024-07-09-16:19:48.156107] 'make' '-j16'
==> [2024-07-09-16:19:48.204159] 'make' '-j16' 'clean'
PASSED: RocmCmake::test_cmake
==> [2024-07-09-16:19:48.249513] Completed testing
==> [2024-07-09-16:19:48.249654] 
====================== SUMMARY: rocm-cmake-5.4.3-245fk42 =======================
RocmCmake::test_cmake .. PASSED
```

**Review by tldahlgren — APPROVED** [review]

**Pull request merged**